debug-dump/src/Debug/Trace/Dump.hs

```diff
@@ -0,0 +1,23 @@
+module Debug.Trace.Dump
+    ( traceFile
+    , traceFileBSL
+    , traceFileText
+    ) where
+
+import qualified Data.ByteString.Lazy as BSL
+import           Data.Functor
+import qualified Data.Text            as T
+import qualified Data.Text.IO         as TIO
+import           System.IO.Unsafe     (unsafePerformIO)
+
+traceFileBSL :: FilePath -> BSL.ByteString -> a -> a
+traceFileBSL fp bytes x = unsafePerformIO $
+    BSL.writeFile fp bytes $> x
```

Comment:
I think `writeFile` is strict and thus this is indeed a correct definition, but I'd like to see a test.

Are you sure you want to overwrite the file rather than to append something to it?

Reply:
That is a good point. I think that the behavior I want downstream is to overwrite the file, but perhaps it should error if the file exists? We might want different versions - one called `traceFileBSL` which errors if the file already exists, and one called `traceFileBSLOverwrite` or the like which would overwrite it.